logging: new mode -l passthrough-tty

giuseppe opened the pull request:
it works in a similar way to passthrough but it allows to be used also on a TTY.

conmon support: https://github.com/containers/conmon/pull/465

Closes: https://github.com/containers/podman/issues/20767

<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Add a new logging mode --passthrough-tty that works like passthrough but can be used with TTYs
```

rhatdan commented:
@giuseppe is this still a draft?

halfline commented:
given `passthrough-tty` got folded into `passthrough` in the conmon merge request, should the same thing happen here?

giuseppe commented:
> given `passthrough-tty` got folded into `passthrough` in the conmon merge request, should the same thing happen here?

personally I think it is better to keep it as a separate mode, to avoid potential security issues with `passthrough` mode.  Now it is explicit that the user expects the std streams to potentially be terminals and accept the risk

Luap99 commented on test/system/035-logs.bats:
```diff
@@ -434,4 +434,26 @@ EOF
+        skip "not running in a tty"
+    fi
+
+    # check if conmon supports -l passthrough-tty, we can drop this check once conmon gets in the CI images.
+    run_podman info -f '{{.Host.Conmon.Path}}'
+    conmon_path="$output"
+
+    if ! $conmon_path -l passthrough --cid foo --cuuid foo --runtime /bin/true; then
+        skip "conmon does not support -l passthrough-tty"
+    fi
```
This is no longer applicable I assume?

Luap99 commented on test/system/035-logs.bats:
```diff
@@ -434,4 +434,26 @@ EOF
+@test "podman run -l passthrough-tty" {
+    skip_if_remote
+
+    if ! tty -s; then
+        skip "not running in a tty"
+    fi
```
looking at CI this seems to always trigger so it is unable to catch any regression in CI thus effectively making the test pointless. Can we setup a tty for this test somehow?

edsantiago commented:
Hi, sorry I haven't been watching this closely, before you spend too much time on this, can you look at `450-interactive.bats`?

giuseppe commented:
> Hi, sorry I haven't been watching this closely, before you spend too much time on this, can you look at `450-interactive.bats`?

thanks, that is surely better!

I've moved the test to `450-interactive.bats` and used the existing infrastructure

edsantiago commented on test/system/450-interactive.bats:
```diff
@@ -102,4 +102,29 @@ function teardown() {
+@test "podman run -l passthrough-tty" {
+    skip_if_remote
+
+    if ! tty -s; then
+        skip "not running in a tty"
+    fi
```
ummmmmm....

giuseppe replied:
dropped now

edsantiago commented on test/system/450-interactive.bats:
```diff
@@ -102,4 +102,29 @@ function teardown() {
+    run_podman info -f '{{.Host.Conmon.Path}}'
+    conmon_path="$output"
+
+    if ! $conmon_path -l passthrough --cid foo --cuuid foo --runtime /bin/true; then
```
Do you know what version of conmon supports this, and where it is in the bodhi pipeline? I just spun up a rawhide VM, and even in updates-testing, I can't get a conmon that supports this. I would like to hold this PR until I can run this test against a truly working conmon.

giuseppe replied:
it is not released yet: https://github.com/containers/conmon/pull/465

Fine by me, let's hold it until the next conmon release

giuseppe commented:
/hold until we have a working conmon in the CI

edsantiago commented:
Ping. We have [new VMs](https://github.com/containers/automation_images/pull/320). Conmon is now `2.1.6+ds1-1` on Debian, and `2:2.1.8-2` on all Fedorae. Is that the required version?

giuseppe commented:
I've rebased and cleaned up the test but unfortunately 2.1.8 is not enough.  We need at least 2.1.9 (which has a regression, so it should be 2.1.10).

TomSweeneyRedHat commented on docs/source/markdown/options/log-driver.md:
```diff
@@ -14,3 +14,5 @@ journald
 container.  It is not allowed with the remote Podman client, including Mac and Windows (excluding WSL2) machines, and on a tty, since it is
 vulnerable to attacks via TIOCSTI.
+
+The **passthrough-tty** driver is the same as **passthrough** except that it allows also to be used on a TTY if the user really wants it.
```
```suggestion
The **passthrough-tty** driver is the same as **passthrough** except that it also allows it to be used on a TTY if the user really wants it.
```

giuseppe replied:
fixed

rhatdan commented:
@giuseppe Could you rebase.

giuseppe commented on test/system/450-interactive.bats:
```diff
@@ -102,4 +102,15 @@ function teardown() {
+
+@test "podman run -l passthrough-tty" {
+    skip_if_remote
+
```
thanks, amended, it did the job!

giuseppe commented:
@containers/podman-maintainers PTAL

Luap99 commented:
/lgtm

Luap99 commented:
/hold cancel
